images/encode.go

```diff
@@ -58,6 +60,40 @@ func EncodeToLimits(bb *bytes.Buffer, img image.Image, bounds DimensionLimits) e
 	return nil
 }
 
+// CompressToFileLimits takes an image.Image and analyses the pixel dimensions, if the longest side is greater
+// than the `longSideMax` image.Image will be resized, before compression begins.
+// Next the image.Image is repeatedly encoded and resized until the data fits within
+// the given FileSizeLimits. There is no limit on the number of times the cycle is performed, the image.Image
+// is reduced to 95% of its size at the end of every round the file size exceeds the given limits.
+func CompressToFileLimits(bb *bytes.Buffer, img image.Image, bounds FileSizeLimits) error {
+	longSideMax := 2000
+
+	// Do we need to do a pre-compression resize?
+	if img.Bounds().Max.X > img.Bounds().Max.Y {
+		// X is longer
+		if img.Bounds().Max.X > longSideMax {
+			img = resize.Resize(uint(longSideMax), 0, img, resize.Bilinear)
+		}
+	} else {
+		// Y is longer or equal
+		if img.Bounds().Max.Y > longSideMax {
+			img = resize.Resize(0, uint(longSideMax), img, resize.Bilinear)
+		}
+	}
+
+	for {
+		err := EncodeToLimits(bb, img, bounds)
+		if err == nil {
+			return nil
+		}
+		if err.Error()[:50] != "image size after processing exceeds max, expect < " {
+			return err
+		}
```

Comment:
Maybe something like [this](https://stackoverflow.com/a/68654658) could be implemented to avoid having to compare the error string?